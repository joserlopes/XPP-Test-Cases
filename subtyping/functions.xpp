type struct1 = struct { #a: int, #b: bool };
type struct2 = struct { #c: int };
type struct3 = struct { #a: int };
let f: struct1 -> struct2 = fn a: struct3 => {
    { #c = a.#a + 2, #d = 10 }
};
f({ #a = 10, #b = true });;

type struct1 = struct { #a: int, #b: bool };
type struct2 = struct { #c: int, #d: int, #e: bool };
type struct3 = struct { #a: int };
let f: struct1 -> struct2 = fn a: struct3 => {
    { #c = a.#a + 2, #d = 10 }
};
f({ #a = 10, #b = true });;

type struct1 = struct { #a: int, #b: bool };
type struct2 = struct { #c: int };
let f = fn a: struct1 -> struct2 => {
    a({ #a = 1, #b = true })
};
f(fn a: struct { #a: int } => { 
    { #c = a.#a + 1, #d = 10 }
});;
