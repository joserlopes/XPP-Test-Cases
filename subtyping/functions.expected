Value: struct { #c = 12; #d = 10 }
Type: struct { #c: int }

throws TypeCheckError

Value: struct { #c = 2; #d = 10 }
Type: struct { #c: int }
